nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -529,11 +532,13 @@ in {
     };
 
     systemd.services.mastodon-web = {
-      after = [ "network.target" ]
+      after = [ "network.target" "mastodon-init-dirs.service" ]
+        ++ (if databaseActuallyCreateLocally then [ "postgresql.service" ] else [])
+        ++ (if cfg.automaticMigrations then [ "mastodon-init-db.service" ] else []);
+      requires = [ "mastodon-init-dirs.service" ]
         ++ (if databaseActuallyCreateLocally then [ "postgresql.service" ] else [])
-        ++ (if cfg.automaticMigrations then [ "mastodon-init-db.service" ] else [ "mastodon-init-dirs.service" ]);
+        ++ (if cfg.automaticMigrations then [ "mastodon-init-db.service" ] else []);
       description = "Mastodon web";
-      wantedBy = [ "multi-user.target" ];
```

Comment:
I think now nothing gets started at all?

I think we should keep `wantedBy = [ "multi-user.target" ];` for mastodon-web, mastodon-sidekiq and mastodon-streaming. Like [upstream does](https://github.com/mastodon/mastodon/tree/main/dist).

My recommendation was only for mastodon-init-dirs and mastodon-init-db

Reply:
Thanks, fixed.